wcfsetup/install/files/ts/WoltLabSuite/Core/Form/Builder/Field/User.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Data handler for a user form builder field in an Ajax form.
+ *
+ * @author  Matthias Schmidt
+ * @copyright 2001-2020 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Form/Builder/Field/User
+ * @since 5.2
+ */
+
+import Field from "./Field";
+import { FormBuilderData } from "../Data";
+import * as Core from "../../../Core";
+import * as UiItemList from "../../../Ui/ItemList/Static";
+
+class User extends Field {
+  protected _getData(): FormBuilderData {
+    const usernames = UiItemList.getValues(this._fieldId)
+      .map((item) => {
+        if (item.objectId) {
+          return item.value;
+        }
+
+        return null;
+      })
+      .filter((v) => v !== null) as string[];
```

Comment:
```suggestion
    const usernames: string[] = UiItemList.getValues(this._fieldId)
      .map((item) => {
        if (item.objectId) {
          return item.value;
        }

        return null;
      })
      .filter((v) => v !== null);
```
For consistency.

Reply:
No, this does not work because after `map`, the return type is `(string|null)[]` and the `filter` call does not change that.

Reply:
Uh, the `filter()` call removes all elements that equal `null`, so the final result is `string[]`?!

Reply:
```
wcfsetup/install/files/ts/WoltLabSuite/Core/Form/Builder/Field/User.ts:18:11 - error TS2322: Type '(string | null)[]' is not assignable to type 'string[]'.
  Type 'string | null' is not assignable to type 'string'.
    Type 'null' is not assignable to type 'string'.

18     const usernames: string[] = UiItemList.getValues(this._fieldId)
```

Reply:
Okay, it appears to be that the compiler cannot infer the result of the `filter()` call. In hindsight, I don't know why I expected this.

The original code is fine, typecasting it is the only way to yield a sane type.